extensions/undertow/deployment/src/main/java/io/quarkus/undertow/deployment/ServletConfig.java

```diff
@@ -30,15 +32,14 @@ public class ServletConfig {
      * {@literal /foo/bar}</li>
      * </ul>
      */
-    @ConfigItem
-    @ConvertWith(ContextPathConverter.class)
-    Optional<String> contextPath;
+    @WithConverter(ContextPathConverter.class)
+    Optional<String> contextPath();
```

Comment:
The converter needs to be in the `Optional`: `Optional<@WithConverter(ContextPathConverter.class) String>`